### PR TITLE
remove trailing slash for overrideUrlName

### PR DESCRIPTION
**Pull request by jordiprats**
Hi,
This PR is for fixing what could be also considered as a configuration mistake. If you set overrideBaseUrl to an url with a trailing slash as follows:

```
        apiSecurity:
          overrideBaseUrl: http://whatever.com/
        uiSecurity:
          overrideBaseUrl: http://whatever.com/
```

It looks like everything is fine but if you click on the source link on any pipeline you'll get this error message:

```
The request was rejected because the URL contained a potentially malicious String
```

Taking a look at the URL you'll realize there's a double slash after the hostname which leads to this error. On my PR you'll find the code needed for ignoring this final slash: It doesn't even matter if you remove it or add the final slash: the operator won't even bother updating the containers because it's just ignored

kind regards,

### CODE REVIEW

**Comment by jordiprats** [comment]
Regading the failed build looks like it's something unrelated to the code:

```

Run docker login -u  -p "" registry.redhat.io
"docker login" requires at most 1 argument.
See 'docker login --help'.

Usage:  docker login [OPTIONS] [SERVER]

Log in to a Docker registry
Error: Process completed with exit code 1.
```

File: pkg/deploy/spindeploy/expose_service/transform.go [hunk 1]
```diff
@@ -84,8 +84,14 @@ func (t *exposeTransformer) setStatusAndOverrideBaseUrl(ctx context.Context, ser
 
 // findStatusUrl returns the overrideBaseUrl or load balancer url, indicating if it came from overrideBaseUrl
 func (t *exposeTransformer) findStatusUrl(ctx context.Context, serviceName string, overrideUrlName string) (string, bool, error) {
+	var overrideUrlName_sanitized string
 	// ignore error, overrideBaseUrl may not be set in hal config
-	statusUrl, _ := t.svc.GetSpinnakerConfig().GetHalConfigPropString(ctx, overrideUrlName)
+	if overrideUrlName[len(overrideUrlName)-1:] == "/" {
```
Comment (dmyerscough) [inline pkg/deploy/spindeploy/expose_service/transform.go:89]: You should leverage `strings` and do `strings.HasPrefix(overrideUrlName, "/")`

```
if strings.HasPrefix(overrideUrlName, "/") {
    overrideUrlName = overrideUrlName[:len(overrideUrlName) - 1]
}
```

You could drop the else and the additional variable.

**Comment by jordiprats** [comment]
Good point, although it is actually [HasSuffix: tests whether the string s ends with suffix.](https://golang.org/pkg/strings/#HasSuffix)

**Comment by jordiprats** [comment]
This change is quite straightforward and it's going to help users avoid getting stuck with this issue. Can someone take a look?

**Review by ichi0915 — APPROVED** [review]

**Comment by ichi0915** [comment]
Thank you for this, I don't have permissions to merge this until the actions have passed let me see who can help us get this merge

**Comment by ichi0915** [comment]
@jordiprats can you help us rebasing to the latest commits in master or maybe create an empty commit just to re-run the actions and see the output.

**Comment by jordiprats** [comment]
@ichi0915 sorry I messed up with this PR, I have created a new one with the change rebased: #267

**Pull request merged**